src/components/headerSearch/headerSearch.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from "react";
+import Mark from "mark.js";
+import './headerSearch.css';
+import { IonButton, IonButtons, IonIcon, IonInput, IonItemDivider } from "@ionic/react";
+import { arrowDown, arrowUp } from 'ionicons/icons';
+
+interface Props {
+  doc: any
+}
+
+const HeaderSearch: React.FC<Props> = ({doc}) => {
+  const [needle, setNeedle] = useState<string>('');
+  const [marks, setMarks] = useState<NodeListOf<HTMLElement>>(document.querySelectorAll('mark'));
+  const [currentIndex, setCurrentIndex] =  useState<number>(0);
+  let mark: any;
+
+  useEffect(() => {
+    search()
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [needle]); 
+
+  const jumpTo = () => {
+    if (marks.length > 0) {
+      const currentMark = marks[currentIndex];
+      marks.forEach(mark => mark.classList.remove('current'));
+      if (currentMark) {
+        currentMark.classList.add('current');
+        currentMark.scrollIntoView({
+          behavior: "smooth",
+          inline: "start"
+        });
+      }
+    }
+  }
+
+  const jumpThrough = () => {
+    if (marks.length > 0) {
+      if (currentIndex < 0) {
+        setCurrentIndex(marks.length - 1);
+      }else if (currentIndex > marks.length - 1) {
+        setCurrentIndex(0);
+      }
+      jumpTo();
+    }
+  }
+  const jumpToNext = () => {
+    setCurrentIndex(currentIndex + 1);
+  }
+  const jumpToPrevious = () => {
+    setCurrentIndex(currentIndex - 1);
+  }
+    
+  useEffect(() => {
+    jumpThrough()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentIndex])
+
+  const search = () => {
+    setCurrentIndex(0)
+
+    if (needle && needle !== '') {
+      if (!mark) {
+        if (doc) {
+          mark = new Mark(doc);
+        }
+      }
+      // unmark existing marks
+      mark.unmark({
+        done: () => markResults()
+      });
+    } else {
+      if (mark) {
```

Comment:
This is always null. Try searching for a string, and then backspacing until there are no search characters -- the final mark doesn't get removed. I would solve this by making this a class so you can stash the mark, but there might be other ways.